src/modules/Instabug.ts

```diff
@@ -56,44 +73,198 @@ function reportCurrentViewForAndroid(screenName: string | null) {
   }
 }
 
+function _logFlags() {
+  console.log(
+    `Andrew: init -> {
+     isNativeInterceptionFeatureEnabled: ${isNativeInterceptionFeatureEnabled},
+     hasAPMNetworkPlugin: ${hasAPMNetworkPlugin},
+     shouldEnableNativeInterception: ${shouldEnableNativeInterception}
+    }`,
+  );
+}
+
 /**
  * Initializes the SDK.
  * This is the main SDK method that does all the magic. This is the only
  * method that SHOULD be called.
  * Should be called in constructor of the AppRegistry component
  * @param config SDK configurations. See {@link InstabugConfig} for more info.
  */
-export const init = (config: InstabugConfig) => {
+export const init = async (config: InstabugConfig) => {
+  // Initialize necessary variables
+  isNativeInterceptionFeatureEnabled = await NativeNetworkLogger.isNativeInterceptionEnabled();
+  if (Platform.OS === 'android') {
+    hasAPMNetworkPlugin = await NativeNetworkLogger.hasAPMNetworkPlugin();
+    shouldEnableNativeInterception =
+      config.networkInterceptionMode === NetworkInterceptionMode.native;
+  }
+
+  // Add app state listener to handle background/foreground transitions
+  addAppStateListener(async (nextAppState) => handleAppStateChange(nextAppState, config));
+
+  //Set APM networking flags for the first time
+  setApmNetworkFlagsIfChanged({
+    isNativeInterceptionFeatureEnabled: isNativeInterceptionFeatureEnabled,
+    hasAPMNetworkPlugin: hasAPMNetworkPlugin,
+    shouldEnableNativeInterception: shouldEnableNativeInterception,
+  });
+
+  // Perform platform-specific checks and update interception mode
+  handleNetworkInterceptionMode(config);
+
+  // Log the current APM network flags and initialize Instabug
+  _logFlags();
+
+  // call Instabug native init method
+  initializeNativeInstabug(config);
+
+  // Set up error capturing and rejection handling
   InstabugUtils.captureJsErrors();
   captureUnhandledRejections();
 
-  // Default networkInterceptionMode to JavaScript
+  _isFirstScreen = true;
+  _currentScreen = firstScreen;
+
+  reportCurrentViewForAndroid(firstScreen);
+  setTimeout(() => {
+    if (_currentScreen === firstScreen) {
+      NativeInstabug.reportScreenChange(firstScreen);
+      _currentScreen = null;
+    }
+  }, 1000);
+};
+
+/**
+ * Handles app state changes and updates APM network flags if necessary.
+ */
+const handleAppStateChange = async (nextAppState: AppStateStatus, config: InstabugConfig) => {
+  if (_currentAppState.match(/inactive|background/) && nextAppState === 'active') {
```

Comment:
I don't think we need a regex for this kind of comparison, we can change this to something like:
```suggestion
  if (['inactive', 'background'].includes(_currentAppState) && nextAppState === 'active') {
```

Reply:
Using regex in this scenario is likely more readable and eliminates the need for a runtime-created list like `['inactive', 'background']`. Additionally, regex comparison is significantly faster than searching through a list.

@a7medev what do you think?

Reply:
@AndrewAminInstabug I have some points regarding this:
1. `String.match(regex)` doesn't actually return a `boolean`, it returns an array of the matched values and their indices, so the check should look `_currentAppState.match(/inactive|background/).length > 0`.
2. As for the readability point, I don't think regular expressions are that readable, besides adding the `.length > 0` would make it even more unreadable, though it's still up to personal preference of course.
3. Performance-wise, regular expression matching is almost 3x slower than `Array.includes` with 2 array elements and 10x slower than using a normal `||` with equality checks, here are the numbers I got when I benchmarked this for more visibility:
```
Regex took 10.399999976158142ms
Array.includes took 3.100000023841858ms
Condition with || took 1.399999976158142ms
```

Reply:
@a7medev thanks a lot for clarification 🙏